eva/expression/comparison_expression.py

```diff
@@ -42,15 +42,20 @@ def evaluate(self, *args, **kwargs):
         lbatch = self.get_child(0).evaluate(*args, **kwargs)
         rbatch = self.get_child(1).evaluate(*args, **kwargs)
 
-        if len(lbatch) != len(rbatch):
-            if len(lbatch) == 1:
-                lbatch.repeat(len(rbatch))
-            elif len(rbatch) == 1:
-                rbatch.repeat(len(lbatch))
-            else:
-                raise Exception(
-                    f"Left and Right batch does not have equal elements: left: {len(lbatch)} right: {len(rbatch)}"
-                )
+        assert len(lbatch) == len(
+            rbatch
+        ), f"Left and Right batch does not have equal elements: left: {len(lbatch)} right: {len(rbatch)}"
+
+        assert self.etype in [
```

Comment:
Won't keeping it in the else clause easier to maintain? Similar to the switch clause in cpp. Future changes won't need to worry about modifying assert. What do you think?

Reply:
Good idea. But, it will be harder to hit the `else` without a synthetic test case (from a coverage standpoint).

Reply:
How about we move the assert condition to the test case? Any update to the logic will fail the test case and will require updating it. 
It is a one-time effort, either we do it in the code using assert or in the test. I prefer test from readability perspective. What do you think?

Reply:
I am not entirely sure if I understand. Can you take care of this change?